### ISSUE TITLE
Installation as Hubitat app hangs

### ISSUE THREAD

**Issue by berkinet** [xAPPO/MQTT#17]
Hubitat C-8 
Platform v 2.3.5.125

I have installed both MQTT app and MQTT Client driver, and both appear under their respective headings under Developer tools.
In the Apps window, I can select Add User App and see the MQTT app. I can then click the app and it seems to start to load, but then hangs.

But, here is where it gets interesting. According to the log, the app is actually running, but failing because it is missing the broker connect information.  I can actually open the app in the log and enter the debugger and restart the app and I get this in the log:
`[dev:1](http://192.168.5.131/logs?tab=past&deviceId=1#)2023-04-15 05:10:11.905 PM[info](http://192.168.5.131/logs?tab=past&deviceId=1#)MQTT> MQTT connect failed attempt:[null], try again in 10 secs
[dev:1](http://192.168.5.131/logs?tab=past&deviceId=1#)2023-04-15 05:10:10.823 PM[info](http://192.168.5.131/logs?tab=past&deviceId=1#)MQTT> Connecting as Hubitat_temporary to MQTT broker null
[dev:1](http://192.168.5.131/logs?tab=past&deviceId=1#)2023-04-15 05:10:10.820 PM[info](http://192.168.5.131/logs?tab=past&deviceId=1#)MQTT> MQTT> client null temporary
[dev:1](http://192.168.5.131/logs?tab=past&deviceId=1#)2023-04-15 05:10:10.807 PM[debug](http://192.168.5.131/logs?tab=past&deviceId=1#)MQTT> Initialise MQTT`

I can click "Info" and see the app settings. But, there is no way to set the broker IP Address.  I have tried restarting Hubitat and still see the same result. No entry in Apps, but still getting the same error messages logged by the app.

UPDATE: I just discovered that I can open the app from the debug screen and configure it. Then, it connects and I see the app in the Apps screen!

**Comment by GledholtHall** [comment]
Great 👍 let me know if you have any further issues

**Comment by berkinet** [comment]
No additional issues for me. However, I would imagine the problem I ran into would be well beyond the capabilities of most people to resolve.  In fact, I am using MQTT to interface to a Indigo smart home server, and several users have reported problems, some sim gave up.

**Comment by GledholtHall** [comment]
Well it’s a configurable option named MQTT Broker.   There is also Jon, the Indigo half of this integration to assist. I don’t use Indigo I’m afraid. I have not had any issues reported for help with Indigo (bar one) so I’m not sure where all those issues were reported.

**Comment by berkinet** [comment]
Yes, I am using Jon’s Indigo plugin.  But, my issue, is, before  even get to Indigo, how do I set this up in Hubitat. When I try to install the Hubitat app, it hangs and never gives me an option to set the broker. That is what I was trying to point out. So, if this is not a bug, how should it properly be done so that I can pass this on to others?

**Comment by GledholtHall** [comment]
What version of the HE MQTT app and MQTT device driver are you using ?

**Comment by berkinet** [comment]
The version here on Github, I installed it yesterday.  Should I be looking somewhere else?

**Comment by GledholtHall** [comment]
Yes you should be using beta 3e pre30+ which I will send you a PM too. It currently has  restricted licence conditions. please note these. There are no instructions yet for this version but I hope it’s pretty self evident. There is an MQTT config option ;-).    
Please uninstall the existing MQTT app and driver. Reboot HE and check they have gone, sometimes you can have several of the older driver loaded.  Then load the new app and driver but only ‘install’ the app, the driver will load itself.

**Comment by GledholtHall** [comment]
Can you send me an email to xappo@ukusa.co.uk and I will reply there

**Comment by krozgrov** [comment]
Is it possible to get the 3e or 4?  Even after deleting the device and rebooting the app just keeps hanging and I am unable to configure my MQTT broker.
